### PR TITLE
Allow rust-analyzer auto publishing

### PR DESCRIPTION
**Pull request by pksunkara**

### CODE REVIEW

File: crates/flycheck/src/lib.rs [hunk 1]
```diff
@@ -54,7 +54,7 @@ impl fmt::Display for FlycheckConfig {
 pub struct FlycheckHandle {
     // XXX: drop order is significant
     sender: Sender<Restart>,
-    thread: jod_thread::JoinHandle,
+    thread: jod_thread::JoinHandle<()>,
```
Comment (pksunkara) [inline crates/flycheck/src/lib.rs:57]:
```
   Compiling flycheck v0.0.0 (/Users/pksunkara/Coding/pksunkara/rust-analyzer/crates/flycheck)
error[E0107]: wrong number of type arguments: expected 1, found 0
  --> /Users/pksunkara/Coding/pksunkara/rust-analyzer/crates/flycheck/src/lib.rs:57:13
   |
57 |     thread: jod_thread::JoinHandle,
   |             ^^^^^^^^^^^^^^^^^^^^^^ expected 1 type argument

error: aborting due to previous error

For more information about this error, try `rustc --explain E0107`.
error: could not compile `flycheck`.
```

I was getting these issues when compiling the crates as path dependencies of another lib. Not sure why `cargo test` isn't failing.

Comment (matklad) [reply crates/flycheck/src/lib.rs:57]: >I was getting these issues when compiling the crates as path dependencies of another lib. Not sure why cargo test isn't failing.

You have outdated Cargo.lock in that context.

Comment (matklad) [reply crates/flycheck/src/lib.rs:57]: That is, let's stick to new jod_thread here, and remove `()`

Comment (pksunkara) [reply crates/flycheck/src/lib.rs]: Done

**Comment by matklad** [comment]
Hm, publish worflow takes 26 minutes... I think this is because it does `O(N^2)` compilations for verification. Let's perhaps pass `--no-verify` flag to avoid this?

**Comment by pksunkara** [comment]
We could, but I had some issues a few months when using that flag, so I am very hesitant to do it. Is there any issue if the workflows takes too long? It is not a dependency of anything.

**Comment by matklad** [comment]
No specific issues, just slight uneasiness at burning electricity for repeatedly doing the same work :)

**Comment by pksunkara** [comment]
Yeah, I would skip it if I could but I encountered many issues while doing that for chalk.

**Comment by pksunkara** [comment]
Renaming doesn't look like it will work with rust-analyzer because it has binary. https://github.com/rust-analyzer/rust-analyzer/runs/1022483250?check_suite_focus=true#step:5:1618

**Comment by MikailBag** [comment]
Maybe pass --no-verify when publishing all crates except the first one?
That way verification will still run, but only once.

**Comment by pksunkara** [comment]
No, the first crate is `ra_ap_arena` which means verifying it verifies only that part of the code.

**Comment by pksunkara** [comment]
So, I looked around and there is no way to fix this issue. We can compromise and allow the auto publish to publish directly to `rust-analyzer`. Any other suggestions are welcome.

**Comment by matklad** [comment]
Seems like this should also be fixable by `--no-verify` flag

**Comment by pksunkara** [comment]
But won't we have trouble when we are downloading the crate to use as library?

**Comment by matklad** [comment]
no, we won't compile the binary in that case. Although, hm, that *would* prevent `cargo install ra_ap_rust_analyzer` from just working, and that would be a nice bonus if we could make it work...

**Comment by matklad** [comment]
I'd probably just `sed /use rust_analyzer/use ra_ap_rust_analyzer/` for files in the src/bin.

**Comment by matklad** [comment]
Hm, alternatively, yeah, let's maybe publish this one crate without `ra_ap` prefix?

**Comment by pksunkara** [comment]
I think both methods work. I would probably lean towards sed.

**Comment by matklad** [comment]
Yeah, I have no strong feelings either way

**Pull request merged**